### ISSUE TITLE
How to use CodeMirror modes?

### ISSUE THREAD

**Issue by dominik-beste** [mgechev/angular2-seed#760]
I've been trying to install CodeMirror and the *xml* and *htmlmixed* modes. I managed to get CodeMirror imported correctly into the project, but don't now how to do it for the modes.

I installed CodeMirror with npm and installed the typings for it. Then in `project.config.ts` I added this line:
```
(<any>this.SYSTEM_CONFIG_DEV.paths)['codemirror'] =
    `${this.APP_BASE}node_modules/codemirror/lib/codemirror`;
```
and for the CSS file added this entry to `additional_deps`:
```
{src: 'codemirror/lib/codemirror.css', inject: true}
```
To use it in a component, I import it as:
```
import * as CodeMirror from 'codemirror';
```
That works.

Once I have CodeMirror imported, I need to execute the code that is in the *mode* files so they extend CodeMirror. For example, in `node_modules/mode/xml/xml.js` there is the line:
```
CodeMirror.defineMode("xml", function(editorConf, config_) { ... }
```
Which relies on the CodeMirror object to exist.

How can I do this, or is there a better way all together for using CodeMirror?

For development, I got it to work, by just including the scripts in index.html, but for production that doesn't work.

**Comment by d3viant0ne** [comment]
@dominik-beste - I've never messed with it but i'm always town to tinker with something new so i'll try it on a branch of the base seed.

For the issue with the production build, can you post the error output if any or the behavior of it "not working"

I want to make sure when it doesn't work locally, it's failing the same way.

**Comment by dominik-beste** [comment]
Thanks for your reply. I got it working for production as well now, but without using the typings. So if I inject all required files as:
```
let additional_deps: InjectableDependency[] = [
             {src: 'codemirror/lib/codemirror.js', inject: 'libs'},
             {src: 'codemirror/mode/xml/xml.js', inject: 'libs'},
             {src: 'codemirror/mode/htmlmixed/htmlmixed.js', inject: 'libs'},
             ...
        ];
```
in `project.config.js`, I can use them in the components like:
```
declare var CodeMirror: any;
```

However, I would like to use CodeMirror with typings. I can do this for CodeMirror, by installing the typings and then use it within the component as:
```
import * as CodeMirror from 'codemirror';
```
This creates a CodeMirror object on the scope of the component. So to use the CodeMirror modes, they will have to be loaded on the same scope after CodeMirror has been initialised. I tried loading the modes the same way, i.e:
```
(<any>this.SYSTEM_CONFIG_DEV.paths)['codemirror-xml'] =
    `${this.APP_BASE}node_modules/codemirror/mode/xml/xml`;
```
and
```
import * as CodeMirrorXml from 'codemirror-xml';
```
but that didn't work.

If I import CodeMirror in the component and load the modes using `additional_deps`, I get an error when it loads the modes, as they rely on CodeMirror object to exist on the global scope.

CodeMirror also works without loading the modes, but I don't get additional features I need from the modes, such as syntax highlighting.

Thanks for your help so far. I can send you the code for the working configuration (without the typings) if that would help.
